torch/csrc/autograd/functions/convolution.cpp

```diff
@@ -327,6 +365,126 @@ auto ConvBackward::releaseVariables() -> void {
+    if (ggO) {
+      ggO = std::make_shared<Add>()->apply({ggO, ggW_term})[0];
+    } else {
+      ggO = ggW_term;
+    }
+  }
+
+  if (ggb) {
+    // View as (1, ggb.size(0), 1, 1...)
+    std::vector<long> new_size(gO->data->sizes().size(), 1);
+    new_size[1] = ggb->data->sizes()[0];
+    auto ggb_contiguous = std::make_shared<Clone>()->apply({ggb})[0];
+    auto ggb_view = std::make_shared<View>(new_size)->apply({ggb_contiguous})[0];
+
+    // Expand 
+    auto ggb_expanded = std::make_shared<Expand>(gO->data->sizes())->apply({ggb_view})[0];
+
+    if (ggO) {
+      ggO = std::make_shared<Add>()->apply({ggO, ggb_expanded})[0];
+    } else {
+      ggO = ggb_expanded;
+    }
+  }
+
+  // Compute gW = conv(gO, ggI)
+  std::shared_ptr<Variable> gW = nullptr;
+  if (ggI) {
+    // Modified params with correct padding
+    ConvParams gw_conv_params(*this);
+    auto weight_size = weight->data->sizes();
+    std::vector<long> kernel_size(weight_size.begin() + 2, weight_size.end());
+    auto input_size = ggI->data->sizes();
+    std::vector<long> input_shape(input_size.begin() + 2, input_size.end());
+    for(size_t i=0; i<gw_conv_params.padding.size(); ++i) {
+      // Formula for conv output size before the floor operation
+      auto out_size = float(input_shape[i] + 2 * gw_conv_params.padding[i] +
+                      gw_conv_params.dilation[i] * (kernel_size[i] - 1) - 1) /
+                      gw_conv_params.stride[i] + 1;
+      if (floorf(out_size) != out_size) {
+        // TODO: narrow ggI here to ignore these elements?
+        throw std::runtime_error("Some input elements have been lost during ConvForward"
+        " (see documentation for the Conv layer) so ConvBackwardBackward cannot be used."
+        " Resize the input so that no element is lost to be able to use ConvBackwardBackward.");
+      }
+      auto tmp = gw_conv_params.dilation[i];
+      gw_conv_params.dilation[i] = gw_conv_params.stride[i];
+      gw_conv_params.stride[i] = tmp;
+    }
+
+    // Transpose gO and ggI to accumulate over batch
+    auto gOt = std::make_shared<Transpose>(0, 1)->apply({gO})[0];
+    auto ggIt = std::make_shared<Transpose>(0, 1)->apply({ggI})[0];
+
+    // Compute conv
+    auto gWt = std::make_shared<ConvForward>(gw_conv_params)->apply({ggIt, gOt, nullptr})[0];
+
+    // Transpose gW to match chan_in and chan_out
+    gW = std::make_shared<Transpose>(0, 1)->apply({gWt})[0];
+  }
+
+  // Compute gI = convT(gO, ggW)
+  std::shared_ptr<Variable> gI = nullptr;
+  if (ggW) {
+    // select conv tranpose and swap stride and dilation
+    ConvParams gi_conv_params(*this);
+    gi_conv_params.transposed = true;
+    for(size_t i=0; i<gi_conv_params.padding.size(); ++i) {
+      if (gi_conv_params.stride[i] != 1) {
+        // TODO: Remove this when transpose dilated is fixed
+        throw std::runtime_error("Setting non-zero ggW for ConvBackwardBackward would require"
+        " using a dilated transpose convolution which is not supported.");
```

Comment:
Well ggW is the user provided gradient.. But i agree that this can be confusing.
Should i call it `argument 1 of ConvBackwardBackward` ?